sceptre/plan/actions.py

```diff
@@ -873,3 +877,97 @@ def _get_cs_status(self, change_set_name):
             return StackChangeSetStatus.DEFUNCT
         else:  # pragma: no cover
             raise Exception("This else should not be reachable.")
+
+    @add_stack_hooks
+    def fetch_remote_template(self):
+        """
+        Returns the Template for the remote Stack
+
+        :returns: the template body.
+        :rtype: str
+        """
+        self.logger.debug("%s - Fetching remote template", self.stack.name)
+
+        response = self.connection_manager.call(
+            service="cloudformation",
+            command="get_template",
+            kwargs={
+                "StackName": self.stack.external_name
+            }
+        )
+
+        return response.get("TemplateBody")
```

Comment:
So weird thing I found: Sometimes this returns a `dict` and sometimes this returns a string. https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation.html#CloudFormation.Client.get_template
![image](https://user-images.githubusercontent.com/77296393/132390003-63ae8058-dc23-446f-a823-055646910099.png)


Most of the time, it returns a dict, actually. But then sometimes, it returns a string, such as when the template has a transform on it, like a SAM template.

Reply:
(that last bit isn't documented behavior, but it _did_ blow up my own diff command until I accounted for it)

Reply:
That's interesting, will have to test that out for myself 😄

Reply:
I think we should use TemplateStage="Original". I say this because the default is to use the processed template, which includes the post-transformed template when there IS a transform. This means that if you have a template with a transform, like a SAM template, your diff will be off, even if you haven't changed the template at all, since the templates we',re generating are all the "original". 

I even played around with aws-sam-translator to translate any Sam templates pre-diff... And there STILL was a difference, since the translator will give auto-generated resources (like API gateways) unique names every time.